hathor/consensus.py

```diff
@@ -1006,6 +1009,7 @@ def add_voided_by(self, tx: Transaction, voided_hash: bytes) -> bool:
             if tx2.is_block:
                 assert isinstance(tx2, Block)
                 self.context.block_algorithm.mark_as_voided(tx2)
+                tx2.storage.update_best_block_tips_cache(None)
```

Comment:
Why do you need this change? Was it a bug?

Reply:
Yes, the best block tips cache wasn't being updated and it was returning the wrong best block in a specific situation. I created this issue to improve our testing tools: https://github.com/HathorNetwork/hathor-core/issues/491.